components/script/dom/bindings/structuredclone.rs

```diff
@@ -7,13 +7,77 @@
 
 use dom::bindings::error::{Error, Fallible};
 use dom::globalscope::GlobalScope;
-use js::jsapi::{HandleValue, MutableHandleValue};
-use js::jsapi::{JSContext, JS_ReadStructuredClone, JS_STRUCTURED_CLONE_VERSION};
-use js::jsapi::{JS_ClearPendingException, JS_WriteStructuredClone};
+use js::jsapi::{Handle, HandleObject, HandleValue, MutableHandleValue};
+use js::jsapi::{Heap, JSContext};
+use js::jsapi::{JSStructuredCloneCallbacks, JSStructuredCloneReader, JSStructuredCloneWriter};
+use js::jsapi::{JS_ClearPendingException, JSObject, JS_ReadStructuredClone};
+use js::jsapi::{JS_STRUCTURED_CLONE_VERSION, JS_WriteStructuredClone};
+use js::jsapi::{MutableHandleObject, TransferableOwnership};
 use libc::size_t;
+use std::os::raw;
 use std::ptr;
 use std::slice;
 
+
+#[allow(unused_variables, dead_code)]
+unsafe extern "C" fn read_callback(cx: *mut JSContext,
+                           r: *mut JSStructuredCloneReader,
+                           tag: u32,
+                           data: u32,
+                           closure: *mut raw::c_void) -> *mut JSObject {
```

Comment:
nit: indent each subsequent line to match the first argument (in all new functions in this file).